sdk/ai/ai-inference-rest/samples-dev/streamingToolCall.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to use tool calls with streaming chat completions.
+ *
+ * @summary use tool call with streaming chat completions.
+ */
+
+import ModelClient, { ChatCompletionsToolCallOutput } from "@azure-rest/ai-inference";
+import { createSseStream } from "@azure/core-sse";
+import { DefaultAzureCredential } from "@azure/identity";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+import { IncomingMessage } from "http";
+dotenv.config();
+
+// You will need to set these environment variables or edit the following values
+const modelEndpoint = process.env["MODEL_ENDPOINT"] || "<endpoint>";
+
+interface EventData {
```

Comment:
any chance this type can be added to the library itself?

Reply:
This is the [StreamingChatCompletionsUpdate](https://github.com/Azure/azure-rest-api-specs/blob/c5959244eecc034bd8595f020fdde6d460550943/specification/ai/ModelClient/models/chat_completions.tsp#L167C7-L167C37) in the TypeSpec, but it doesn't get genned out in the RLC. I try to regen today and add those classes to the public model list.

Reply:
/cc @MaryGao @joheredi

Reply:
> @glharper add javascript in addition to python in client.tsp: @@access(AI.Model.StreamingChatChoiceUpdate, Access.public, "python"); @@access(AI.Model.StreamingChatCompletionsUpdate, Access.public, "python");

@dargilco See [this line](https://github.com/Azure/azure-rest-api-specs/blob/26bb95d75c97f331449ca1d2ed8b36727180a4af/specification/ai/ModelClient/client.tsp#L46) and the one below in my branch. No dice genning from that.